app/src/main/java/com/github/se/eduverse/model/MediaCacheManager.kt

```diff
@@ -112,18 +155,43 @@ class MediaCacheManager(
     return file.delete()
   }
 
+  @OptIn(UnstableApi::class)
   fun cleanCache() {
     val currentTime = System.currentTimeMillis()
     val cacheDir = context.cacheDir
+
     cacheDir.listFiles()?.forEach { file ->
-      if (currentTime - file.lastModified() > CACHE_LIFETIME_MS) {
-        file.delete()
+      val timestamp = file.name.split("_").firstOrNull()?.toLongOrNull()
+      if (timestamp != null) {
+        val age = currentTime - timestamp
+        Log.d("CACHE_CLEAN", "File: ${file.name}, Age: $age ms")
+
+        if (age > CACHE_LIFETIME_MS) {
+          if (file.delete()) {
+            Log.d("CACHE_CLEAN", "Deleted expired file: ${file.name}")
+          } else {
+            Log.e("CACHE_CLEAN", "Failed to delete file: ${file.name}")
+          }
+        }
+      } else {
+        Log.e("CACHE_CLEAN", "Invalid file name, cannot extract timestamp: ${file.name}")
       }
     }
   }
 
+  @OptIn(UnstableApi::class)
   fun hasCachedFiles(): Boolean {
+    val currentTime = System.currentTimeMillis()
     val cacheDir = context.cacheDir
-    return cacheDir.listFiles()?.isNotEmpty() ?: false
+    val hasRecentFiles =
+        cacheDir.listFiles()?.any { file ->
+          val timestamp = file.name.split("_").firstOrNull()?.toLongOrNull()
+          val isRecent = timestamp != null && currentTime - timestamp <= CACHE_LIFETIME_MS
+          Log.d("CACHE_CHECK", "File: ${file.name}, Is recent: $isRecent")
+          isRecent
+        } ?: false
+
+    Log.d("CACHE_CHECK", "Has cached files: $hasRecentFiles")
+    return hasRecentFiles
```

Comment:
The logic effectively checks if any file in the directory is "recent" by comparing the timestamp against CACHE_LIFETIME_MS.
Logs the name of each file and whether it qualifies as "recent," providing clarity during debugging.
Safely handles the case where listFiles() returns null by using the Elvis operator (?: false).
Returns as soon as a recent file is found, optimizing performance by avoiding unnecessary iterations.